lib/Doctrine/ORM/Cache/DefaultEntityHydrator.php

```diff
@@ -73,8 +73,15 @@ public function buildCacheEntry(ClassMetadata $metadata, EntityCacheKey $key, $e
             }
 
             if ( ! isset($assoc['cache']) || ! ($assoc['type'] & ClassMetadata::TO_ONE)) {
+                $associatedEntity = $data[$name];
                 unset($data[$name]);
 
+                if ($this->uow->isInIdentityMap($associatedEntity)) {
```

Comment:
I'm not an expert but I think some kind of proxy should be created to allow traversal of the association (that will obvisouly trigger another request).